### PR TITLE
user interface: translations

### PR DESCRIPTION
**Pull request by iGormilhit**
* NEW Updates the translations.
* FIX Ignores the angular builds when extracting messages to translate.
* FIX Adds a favicon for apple touche devices.

Signed-off-by: Igor Milhit <igor.milhit@rero.ch>

-----

### How to test

- `pipenv run invenio collect` (for the apple favicon), but you'll need to build the angluar app, so `./scripts/bootstrap` is the way...
- check the user interface (public and professional) with different locales

### CODE REVIEW

File: ui/admin/src/assets/i18n/en.json [hunk 1]
```diff
@@ -118,13 +121,13 @@
   "Street": "Street",
   "City": "City",
   "No result found.": "No result found.",
-  "monday": "monday",
```
Comment (jma) [inline ui/admin/src/assets/i18n/en.json]: I'm not sure that the translation mechanism should be used to apply an uppercase. (this has been probably already used at other places). We can use CSS rules for this.

File: ui/admin/src/assets/i18n/en.json [hunk 2]
```diff
@@ -159,19 +162,18 @@
   "Checkout duration": "Checkout duration",
   "Checkout duration is required.": "Checkout duration is required.",
   "Circulation Policy": "Circulation Policy",
-  "Number renewals": "Number renewals",
-  "Checkout duration must be great than 0.": "Checkout duration must be great than 0.",
+  "Number renewals": "Number of renewals",
```
Comment (jma) [inline ui/admin/src/assets/i18n/en.json]: Should be changed in the original string?

File: ui/admin/src/assets/i18n/en.json [hunk 3]
```diff
@@ -99,17 +99,20 @@
   "has 1 patron attached": "has 1 patron attached",
   "has # patrons attached": "has # patrons attached",
   "The default record cannot be deleted": "The default record cannot be deleted",
-  "The recording contains settings": "The recording contains settings",
+  "The recording contains settings": "The record contains settings",
```
Comment (jma) [inline ui/admin/src/assets/i18n/en.json]: Should be changed in the original string?

File: ui/admin/src/assets/i18n/en_US.json [hunk 4]
```diff
@@ -100,16 +100,19 @@
...
   "Record deleted.": "Record deleted.",
+  "no location": "no location",
   "Address": "Address",
   "Email": "Email",
   "Opening hours and holidays:": "Opening hours and holidays:",
   "{{ opening_hour.day }}:": "{{ opening_hour.day }}:",
   "Closed": "Closed",
   "locations": "locations",
-  "no location": "no location",
+  "{{ 'no location' | translate }}": "{{ 'no location' | translate }}",
```
Comment (jma) [inline ui/admin/src/assets/i18n/en_US.json]: not good!

File: ui/admin/src/assets/i18n/fr.json [hunk 5]
```diff
@@ -10,7 +10,7 @@
   "ITEM_ON_LOAN": "en prêt",
   "ITEM_RETURNED": "retourné",
   "ITEM_IN_TRANSIT_FOR_PICKUP": "en transit pour retrait",
-  "ITEM_IN_TRANSIT_TO_HOUSE": "en transit à son origine",
+  "ITEM_IN_TRANSIT_TO_HOUSE": "ITEM_IN_TRANSIT_TO_HOUSE",
```
Comment (jma) [inline ui/admin/src/assets/i18n/fr.json]: why this is not translated?

File: ui/admin/src/assets/i18n/fr.json [hunk 6]
```diff
@@ -100,17 +100,22 @@
...
+  "no location": "pas de dépôt",
   "Address": "Adresse",
   "Email": "E-mail",
-  "Opening hours and holidays:": "Heures d'ouverture et vacances",
+  "Opening hours and holidays:": "Heures d'ouverture et congés",
+  "{{ opening_hour.day }}:": "{{ opening_hour.day }}:",
   "Closed": "Fermé",
   "locations": "dépôt",
-  "no location": "pas de dépôt",
+  "{{ 'no location' | translate }}": "{{ 'no location' | translate }}",
```
Comment (jma) [inline ui/admin/src/assets/i18n/fr.json]: not good

File: ui/admin/src/assets/i18n/it.json [hunk 7]
```diff
@@ -23,107 +23,112 @@
...
+  "Add": "Add",
+  "Edit": "Edit",
+  "items": "items",
+  "no item": "no item",
+  "Record deleted.": "Record deleted.",
+  "no location": "no location",
+  "Address": "Address",
+  "Email": "Email",
+  "Opening hours and holidays:": "Opening hours and holidays:",
+  "{{ opening_hour.day }}:": "{{ opening_hour.day }}:",
```
Comment (jma) [inline ui/admin/src/assets/i18n/it.json]: bug?

File: ui/admin/src/assets/i18n/it.json [hunk 8]
```diff
@@ -23,107 +23,112 @@
...
+  "Closed": "Closed",
+  "locations": "locations",
+  "{{ 'no location' | translate }}": "{{ 'no location' | translate }}",
```
Comment (jma) [inline ui/admin/src/assets/i18n/it.json]: bug?

File: ui/admin/src/assets/i18n/en.json (hunk 1 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/en.json]: Yes, in some place the CSS rule is a good method, but we already encounters issues with facets. It's then difficult to find the right selector to use `::first-letter`.

File: ui/admin/src/assets/i18n/en.json (hunk 2 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/en.json]: Ok

File: ui/admin/src/assets/i18n/en.json (hunk 3 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/en.json]: Ok

File: ui/admin/src/assets/i18n/en_US.json (hunk 4 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/en_US.json]: Indeed. Corrected.

File: ui/admin/src/assets/i18n/fr.json (hunk 6 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/fr.json]: Corrected.

File: ui/admin/src/assets/i18n/it.json (hunk 7 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/it.json]: Corrected

File: ui/admin/src/assets/i18n/it.json (hunk 8 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/it.json]: Corrected.

File: ui/admin/src/assets/i18n/fr.json (hunk 5 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/fr.json]: I have translated it in English, French and German.

File: ui/admin/src/assets/i18n/en.json [hunk 9]
```diff
@@ -10,13 +10,13 @@
   "ITEM_ON_LOAN": "ITEM_ON_LOAN",
```
Comment (BadrAly) [inline ui/admin/src/assets/i18n/en.json]: Item in circulation or Item checked-out

File: ui/admin/src/assets/i18n/en.json [hunk 10]
```diff
@@ -10,13 +10,13 @@
...
   "ITEM_RETURNED": "ITEM_RETURNED",
   "ITEM_IN_TRANSIT_FOR_PICKUP": "ITEM_IN_TRANSIT_FOR_PICKUP",
```
Comment (BadrAly) [inline ui/admin/src/assets/i18n/en.json]: Item in transit for pick-up

File: ui/admin/src/assets/i18n/en.json [hunk 11]
```diff
@@ -10,13 +10,13 @@
...
-  "ITEM_IN_TRANSIT_TO_HOUSE": "ITEM_IN_TRANSIT_TO_HOUSE",
+  "ITEM_IN_TRANSIT_TO_HOUSE": "Item in transit to owning library",
   "ITEM_AT_DESK": "ITEM_AT_DESK",
```
Comment (BadrAly) [inline ui/admin/src/assets/i18n/en.json]: Item at desk

File: ui/admin/src/assets/i18n/en.json [hunk 12]
```diff
@@ -10,13 +10,13 @@
   "ITEM_ON_LOAN": "ITEM_ON_LOAN",
   "ITEM_RETURNED": "ITEM_RETURNED",
```
Comment (BadrAly) [inline ui/admin/src/assets/i18n/en.json]: Item returned or checked-in

File: ui/admin/src/assets/i18n/en.json (hunk 9 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/en.json]: Done.

File: ui/admin/src/assets/i18n/en.json (hunk 10 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/en.json]: Done.

File: ui/admin/src/assets/i18n/en.json (hunk 11 above)
Comment (iGormilhit) [reply ui/admin/src/assets/i18n/en.json]: Done.

**Pull request merged**